### PR TITLE
Feature/swift 4

### PR DESCRIPTION
**Pull request by patgoley**

### CODE REVIEW

File: src/nodekit/NKScripting/util/NKLogging.swift [hunk 1]
```diff
@@ -174,29 +173,29 @@ public class NKLogging {
...
             
-            filter = Filter(upto: .Notice)
+            filter = Filter(upto: .notice)
             
         #endif
         
         let format = format ?? "$((Time)(lcl)) $(Facility) <$((Level)(char))>: $(Message)"
         
-        asl_add_output_file(client, STDERR_FILENO, format, "sec", filter.rawValue, 1)
+        _ = asl_add_output_file(client, STDERR_FILENO, format, "sec", filter.rawValue, 1)
```
Comment (travis-j-smith) [inline src/nodekit/NKScripting/util/NKLogging.swift]: Is there no way to annotate this method with `@discardableResult`?  I'm not familiar with the syntax at the top of this file so perhaps it's not possible.

Comment (patgoley) [reply src/nodekit/NKScripting/util/NKLogging.swift]: I am not sure, we're using this _sil_gen_name keyword to declare these functions and I think they are linked at runtime, so I think I have to leave the signature how it is. Should be fine either way.

File: src/nodekit/NKScriptingLite/NKJSContext.swift [hunk 2]
```diff
@@ -158,24 +148,24 @@ extension NKJSContext: NKScriptContext {
...
-            obj = val as? NSNumber ?? val.nonretainedObjectValue
+            obj = val as? NSNumber ?? val.nonretainedObjectValue as AnyObject?
        
         }
 
         
         if let s = obj as? String {
          
-            let d = try? NSJSONSerialization.dataWithJSONObject([s], options: NSJSONWritingOptions(rawValue: 0))
+            let d = try? JSONSerialization.data(withJSONObject: [s], options: JSONSerialization.WritingOptions(rawValue: 0))
```
Comment (travis-j-smith) [inline src/nodekit/NKScriptingLite/NKJSContext.swift]: I think you can actually omit the `options` argument.  If I remember correctly it defaults to an empty array, so there's no need to specify a `rawValue` of 0.

**Comment by travis-j-smith** [comment]
Approved

**Pull request merged**